### PR TITLE
fix occasional test error due to custom rspec matcher bug

### PR DESCRIPTION
**Pull request by lazylester**
no production code changed, just rspec matcher and an rspec test added to confirm that the matcher is working as expected

### CODE REVIEW

**Comment by lazylester** [comment]
aargh travis failed on the very bug I fixed!  please standby for a fix to the fix.

**Comment by markottaviani** [comment]
Which existing rspec test is this supposed to help out with?

**Comment by lazylester** [comment]
spec/models/dog_search_spec.rb line 62 "searching for dogs by name with an asc sort".

File: spec/models/dog_search_spec.rb [hunk 1]
```diff
@@ -71,6 +72,32 @@
...
+          else
+            expect(results.map(&:name)).to be_sorted(:ascending)
+          end
+        end
+      end
+
+      context 'confirm postgres compares strings as expected' do
+        let(:bool){ ActiveRecord::Base.connection.execute("select 'Da Vinci' < 'Dallas' as bool").first["bool"]}
+        it "really should return true. when this test fails on TravisCI, we'll know that the postgres sorting has changed" do
+          if ActiveRecord::Base.connection.current_database == "travis_ci_test"
```
Comment (markottaviani) [inline spec/models/dog_search_spec.rb]: Since this tests fails for me too, might need to adjust this to check for Ubuntu 14 instead vs Mac OS.

**Pull request merged**